### ISSUE TITLE
Sklearn pairwise? 

### ISSUE THREAD

**Issue by snakers4** [erikbern/ann-benchmarks#156]
Forgive me for my ignorance, but for a low throughput scenario (i.e. ~10-20 items per second) we used [pairwise cosine](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html) from sklearn which does not suffer from problems that approximate search libraries have (i.e. rebuilding index, adding to index, sharing the index to multiple nodes, etc etc).  

With this you can just minimize your vectors, maybe lower precision and it works decently. The major limitation - you cannot really go beyond 10k - 100k items in the database without search becoming unreasonably slow. The major advantage - you can just store your database as serialized numpy array and add items to it easily.

So my question is, this method was not included because of limited scaling / throughput? Combined with domain heuristics (i.e. you can a priori split your database into 10 chunks) this is a very practical solution.

**Comment by maumueller** [comment]
Hi! Sorry for the super-late reply!

I think our bruteforce implementation here https://github.com/erikbern/ann-benchmarks/blob/master/ann_benchmarks/algorithms/bruteforce.py should basically provide identical performance.
